components/gfx/font.rs

```diff
@@ -218,7 +218,7 @@ impl Font {
     /// Fast path for ASCII text that only needs simple horizontal LTR kerning.
     fn shape_text_fast(&self, text: &str, options: &ShapingOptions, glyphs: &mut GlyphStore) {
         let mut prev_glyph_id = None;
-        for (i, byte) in text.bytes().enumerate() {
+        for (i, byte) in text.to_uppercase().bytes().enumerate() {
```

Comment:
It seems unfortunate for this to allocate.

And the byte is a single byte, so I'm not sure what's wrong with doing it on the fly? Maybe this should not iterate over bytes?

Reply:
I am not sure. 
I just try to follow the  #5938 issue's concept.
But I agree with your point.

Reply:
> However it may not be as easy as upper-casing entire text nodes, we only want to reduce the font size of things that were previously lower-case.

I saw this in the issue. Maybe we don't need to iterate over bytes?